process_sample_data.py

```diff
@@ -20,9 +21,8 @@ class ProcessSampleData():
     """
 
     def __init__(self, path_to_data):
-        self._flow_data_file = open(path_to_data, "r")
-        self._parseData()
-        self._flow_data_file.close()
+        with open(path_to_data, "r") as self._flow_data_file:
```

Comment:
This could be an argument rather than a class member. That's a little more clear, and a little more testable.